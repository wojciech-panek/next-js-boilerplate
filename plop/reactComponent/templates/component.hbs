import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { compose } from 'ramda';

import withIntl from '../../shared/lib/withIntl'; // TODO fix relative import
import { Container } from './{{ camelCase name }}.styles';


export class {{ pascalCase name }} extends PureComponent {
  static propTypes = {
  };

  render() {
    return (
      <Container>
        {{ pascalCase name }} component
      </Container>
    );
  }
}

export default compose(
  withIntl,
)({{ pascalCase name }});

