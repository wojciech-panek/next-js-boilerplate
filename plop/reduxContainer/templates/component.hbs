import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';


export class {{ pascalCase name }} extends PureComponent {
  static propTypes = {
  };

  render() {
    return (
      <div>
        {{ pascalCase name }} component
      </div>
    );
  }
}
