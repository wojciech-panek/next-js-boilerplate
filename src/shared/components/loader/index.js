export { default as Loader } from './loader.component';
