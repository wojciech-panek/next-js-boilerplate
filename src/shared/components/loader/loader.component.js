import React, { PureComponent } from 'react';
import { compose } from 'ramda';

import withIntl from '../../lib/withIntl';
import { Container } from './loader.styles';


export class Loader extends PureComponent {
  static propTypes = {
  };


  render() {
    return (
      <Container>Loading...</Container>
    );
  }
}

export default compose(
  withIntl,
)(Loader);
