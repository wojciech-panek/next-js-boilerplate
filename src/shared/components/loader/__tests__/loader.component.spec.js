import React from 'react';
import { shallow } from 'enzyme';

import { Loader } from '../loader.component';


describe('Loader: Component', () => {
  const defaultProps = {
  };

  const component = (props) => (
    <Loader {...defaultProps} {...props} />
  );

  it('should render correctly', () => {
    const wrapper = shallow(component());
    global.expect(wrapper).toMatchSnapshot();
  });
});
