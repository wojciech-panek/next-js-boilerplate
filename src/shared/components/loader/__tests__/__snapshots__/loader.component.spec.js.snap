// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader: Component should render correctly 1`] = `
<loaderstyles__Container>
  Loading...
</loaderstyles__Container>
`;
